src/main.js

```diff
@@ -128,18 +128,40 @@ const createWindow = async () => {
   });
 };
 
-// This method will be called when Electron has finished
-// initialization and is ready to create browser windows.
-// Some APIs can only be used after this event occurs.
-app.on('ready', createWindow);
-
-app.on('activate', () => {
-  // On OS X it's common to re-create a window in the app when the
-  // dock icon is clicked and there are no other windows open.
-  if (mainWindow === null) {
-    createWindow();
-  }
-});
+// Single instance lock so subsequent instances of the application redirect to
+// the already-open one.
+// Adapted from https://www.electronjs.org/docs/api/app#apprequestsingleinstancelock
+const gotTheLock = app.requestSingleInstanceLock();
+if (!gotTheLock) {
```

Comment:
Nice that they made this easy for us!

Reply:
It really is!  Implementation-wise, it looks like it does ultimately come down to a lock file, which is [managed in the chromium source](https://github.com/electron/electron/blob/ddf3ef0a5fe948fdad4831e29110df25f50882be/chromium_src/chrome/browser/process_singleton_posix.cc#L723).  Although I haven't found the line of source that says so, the lock appears to be unique to our application, probably distinguished by application name in `package.json`.